tron/serialize/filehandler.py

```diff
@@ -50,21 +52,23 @@ def close(self):
 
     def close_wrapped(self):
         """Close only the underlying file handle."""
-        self._fh.close()
-        self._fh = NullFileHandle
+        with self._fh_lock:
+            self._fh.close()
+            self._fh = NullFileHandle
 
     def write(self, content):
         """Write content to the fh. Re-open if necessary."""
-        if self._fh == NullFileHandle:
-            try:
-                self._fh = open(self.name, 'ab')
-            except IOError as e:
-                log.error("Failed to open %s: %s", self.name, e)
-                return
-
-        self.last_accessed = time.time()
-        self._fh.write(maybe_encode(content))
-        self.manager.update(self)
+        with self._fh_lock:
+            if self._fh == NullFileHandle:
+                try:
+                    self._fh = open(self.name, 'ab')
```

Comment:
If write gets called after close, the file handle will not be explicitly closed. Is that okay?

Reply:
Probably not.  Let's print a warning in that case.

Reply:
OK. Another possibility would be to just open the file handle once, in init, and not re-open and write if the file has already been closed.

Reply:
It seems that the idea of this class and `FileHandleManager` below is to close files when they are idle and to reopen them when they're accessed again. If we'll just open a file in `__init__` and don't close it from the manager, we probably could just remove both classes.

Reply:
Yeah, the reason I don't think we need it in this class is that the FileHandleManager just calls `close()` on FileHandleWrappers, which removes the FileHandleWrapper from its cache. So the FileHandleManager re-creates the FileHandleWrapper when a file is accessed again. No one is calling close_wrapped, which leaves the closed wrapper around.

I think that means we can let the FileHandleManager take care of the re-opening logic, and remove it from here. Up to you - we probably won't have too many open file handles around, but it would be nice to keep it clean if it makes sense.